src/app/cookiePreference.service.spec.ts

```diff
@@ -0,0 +1,60 @@
+import {CookiePreferenceService} from "./cookiePreference.service";
+import {CookieService} from "ngx-cookie-service";
+
+describe('CookiePreferenceService', () => {
+
+  const mockCookieService: CookieService = {
+    get: undefined,
+    set: undefined,
+  } as unknown as CookieService;
+
+  it('should show say user has not expressed cookie preference if no cookie preference cookie is set', (done) => {
+    mockCookieService.get = (name) => {
+      expect(name).toBe('cookie-preferences');
+      return '';
+    }
+
+    const service = new CookiePreferenceService(mockCookieService);
+
+    service.userHasExpressedCookiePreference().subscribe({
+      next: value => {
+        expect(value).toBe(false);
+        done();
+      }
+    })
+  })
+
+  it('should show say user not expressed cookie preference if all cookies accepted cookie is set', (done) => {
+    mockCookieService.get = (name) => {
+      expect(name).toBe('cookie-preferences');
+      return '{"agreedToAll": true}';
+    };
+
+    const service = new CookiePreferenceService(mockCookieService);
+
+    service.userHasExpressedCookiePreference().subscribe({
+      next: value => {
+        expect(value).toBe(true);
+        done();
+      }
+    })
+  })
+
+  it('Should set a cookie to show the user accepts all cookies', done => {
+    // @ts-ignore - we don't need to take all the params, only the ones our SUT will actually pass.
+    mockCookieService.set = (name, value: string, duration: number) => {
```

Comment:
realised we we also need to set the cookie path and domain so it can be shared across the site and with the wordpress site.

Reply:
Done in [7634540](https://github.com/thebiggive/donate-frontend/pull/1254/commits/7634540caa741778cf883ddaaac4cdef56093a84)